homeassistant/components/light/__init__.py

```diff
@@ -336,17 +346,21 @@ async def async_handle_light_on_service(light, call):
         if supported_color_modes:
             params.pop(ATTR_WHITE_VALUE, None)
 
-        # Zero brightness: Light will be turned off
-        if params.get(ATTR_BRIGHTNESS) == 0:
-            await light.async_turn_off(**filter_turn_off_params(params))
-        else:
-            await light.async_turn_on(**params)
+        await light.async_turn_on(**params)
```

Comment:
If the resulting brightness is 0, it should still be routed to the async_turn_off call

Reply:
Fixed in https://github.com/home-assistant/core/pull/49376/commits/190402831f40e85f8078560bf1380f79e9486cee